src/keys.ts

```diff
@@ -54,15 +55,27 @@ function parseExtractedKey(key: ExtractedKey, config: Config): TranslationKey {
     const nsSeparatorPos = cleanKey.indexOf(config.nsSeparator);
 
     if (nsSeparatorPos !== -1) {
+      if (key.keyPrefix) {
+        throw new Error(
+          `Do not use the keyPrefix option if you want to use keys with prefixed namespace notation.
+          key: ${cleanKey}
```

Comment:
Nit: indentation is not desired in an error message. Maybe just put the values inline, in parentheses.